Package request: hypridle

**Project description**

Hyprland's idle daemon 

**Metadata**

* homepage URL: https://github.com/hyprwm/hypridle
* source URL: https://github.com/hyprwm/hypridle
* license: BSD 3 clause
* platforms: Linux

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

There is an issue with hypridle currently

<details><summary>Build Log</summary>
<p>

```
> -- Build files have been written to: /build/source/build
       > cmake: enabled parallel building
       > cmake: enabled parallel installing
       > Running phase: buildPhase
       > build flags: -j8 SHELL=/nix/store/5l50g7kzj7v0rdhshld1vx46rf2k5lf9-bash-5.2p26/bin/bash
       > [ 50%] Building CXX object CMakeFiles/hypridle.dir/src/main.cpp.o
       > [ 50%] Building CXX object CMakeFiles/hypridle.dir/src/core/Hypridle.cpp.o
       > [ 50%] Building CXX object CMakeFiles/hypridle.dir/src/helpers/VarList.cpp.o
       > [ 66%] Building CXX object CMakeFiles/hypridle.dir/src/config/ConfigManager.cpp.o
       > [ 83%] Building C object CMakeFiles/hypridle.dir/protocols/ext-idle-notify-v1-protocol.c.o
       > /build/source/src/config/ConfigManager.cpp: In member function 'Hyprlang::CParseResult CConfigManager::postParse()':
       > /build/source/src/config/ConfigManager.cpp:41:44: error: 'class Hyprlang::CConfig' has no member named 'listKeysForSpecialCategory'
       >    41 |     const auto             KEYS = m_config.listKeysForSpecialCategory("listener");
       >       |                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~
       > /build/source/src/core/Hypridle.cpp: In function 'void spawn(const std::string&)':
       > /build/source/src/core/Hypridle.cpp:126:14: warning: ignoring return value of 'ssize_t write(int, const void*, size_t)' declared with attribute 'warn_unused_result' [-Wunused-result]
       >   126 |         write(socket[1], &grandchild, sizeof(grandchild));
       >       |         ~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
       > /build/source/src/core/Hypridle.cpp:133:9: warning: ignoring return value of 'ssize_t read(int, void*, size_t)' declared with attribute 'warn_unused_result' [-Wunused-result]
       >   133 |     read(socket[0], &grandchild, sizeof(grandchild));
       >       |     ~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
       > make[2]: *** [CMakeFiles/hypridle.dir/build.make:76: CMakeFiles/hypridle.dir/src/config/ConfigManager.cpp.o] Error 1
       > make[2]: *** Waiting for unfinished jobs....
       > make[1]: *** [CMakeFiles/Makefile2:83: CMakeFiles/hypridle.dir/all] Error 2
       > make: *** [Makefile:136: all] Error 2
       For full logs, run 'nix log /nix/store/0niw1dkjqrxpyagj7gzjc9jhcwd2sw63-hypridle-unstable-2024-02-17.drv'.
```

This
```
 > /build/source/src/config/ConfigManager.cpp:41:44: error: 'class Hyprlang::CConfig' has no member named 'listKeysForSpecialCategory'
       >    41 |     const auto             KEYS = m_config.listKeysForSpecialCategory("listener");
       >       |                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~
```

</p>
</details>

--- a/pkgs/by-name/hy/hypridle/package.nix
+++ b/pkgs/by-name/hy/hypridle/package.nix
@@ -1,0 +1,45 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, pkg-config
+, cmake
+, wayland
+, wayland-protocols
+, hyprlang
+, sdbus-cpp
+, systemd
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "hypridle";
+  version = "0.1.0";
+
+  src = fetchFromGitHub {
+    owner = "hyprwm";
+    repo = "hypridle";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-0x5R6v82nKBualYf+TxAduMsvG80EZAl7gofTIYtpf4=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+  ];
+
+  buildInputs = [
+    hyprlang
+    sdbus-cpp
+    systemd
+    wayland
+    wayland-protocols
+  ];
+
+  meta = {
+    description = "Hyprland's idle daemon";
+    homepage = "https://github.com/hyprwm/hypridle";
+    license = lib.licenses.bsd3;
+    maintainers = with lib.maintainers; [ iogamaster ];
+    mainProgram = "hypridle";
+    platforms = [ "aarch64-linux" "x86_64-linux" ];
+  };
+})

--- a/pkgs/by-name/hy/hyprlang/package.nix
+++ b/pkgs/by-name/hy/hyprlang/package.nix
@@ -4,15 +4,16 @@
   fetchFromGitHub,
   cmake,
 }:
+
 stdenv.mkDerivation (finalAttrs: {
   pname = "hyprlang";
-  version = "0.3.2";
+  version = "0.4.0";
 
   src = fetchFromGitHub {
     owner = "hyprwm";
     repo = "hyprlang";
     rev = "v${finalAttrs.version}";
-    hash = "sha256-9TT3xk++LI5/SPYgjYX34xZ4ebR93c1uerIq+SE/ues=";
+    hash = "sha256-nW3Zrhh9RJcMTvOcXAaKADnJM/g6tDf3121lJtTHnYo=";
   };
 
   nativeBuildInputs = [cmake];

--- a/pkgs/by-name/hy/hyprlang/package.nix
+++ b/pkgs/by-name/hy/hyprlang/package.nix
@@ -27,5 +27,6 @@ stdenv.mkDerivation (finalAttrs: {
     description = "The official implementation library for the hypr config language";
     license = licenses.gpl3Plus;
     platforms = platforms.linux;
+    maintainers = with maintainers; [ iogamaster ];
   };
 })

--- a/pkgs/by-name/hy/hyprlang/package.nix
+++ b/pkgs/by-name/hy/hyprlang/package.nix
@@ -27,6 +27,6 @@ stdenv.mkDerivation (finalAttrs: {
     description = "The official implementation library for the hypr config language";
     license = licenses.gpl3Plus;
     platforms = platforms.linux;
-    maintainers = with maintainers; [ iogamaster ];
+    maintainers = with maintainers; [ iogamaster fufexan ];
   };
 })

